crux-core/src/crux/api/v2/CruxNode.java

```diff
@@ -0,0 +1,75 @@
+package crux.api.v2;
+
+import clojure.lang.Keyword;
+import clojure.lang.PersistentVector;
+import crux.api.Crux;
+import crux.api.ICruxAPI;
+import crux.api.IndexVersionOutOfSyncException;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static crux.api.v2.Database.database;
+import static crux.api.v2.TxResult.txResult;
+
+public class CruxNode {
+    private final ICruxAPI node;
+
+    private CruxNode(Topology topology) throws IndexVersionOutOfSyncException {
+        this.node = Crux.startNode(topology.toEdn());
+    }
+
+    public static CruxNode cruxNode(Topology topology) throws IndexVersionOutOfSyncException {
+        return new CruxNode(topology);
+    }
+
+    @SuppressWarnings("unchecked")
+    public TxResult submitTx(Iterable<Operation> ops) {
+        PersistentVector txVector = PersistentVector.create();
+        for(Operation op: ops) {
+            txVector = txVector.cons(op.toEdn());
+        }
+
+        Map<Keyword,Object> result = node.submitTx(txVector);
+        Date txTime = (Date) result.get(Keyword.intern("crux.tx/tx-time"));
+        Long txId = (Long) result.get(Keyword.intern("crux.tx/tx-id"));
+        return txResult(txTime, txId);
+    }
+
+    @SuppressWarnings("unchecked")
+    public TxResult submitTx(Operation... ops) { return submitTx(Arrays.asList(ops)); }
+
+    public Database db() {
+        return database(node);
+    }
+
+    public Database db(Date validTime) {
+        return database(node, validTime);
+    }
+
+    public Database db(Date validTime, Date transactionTime) {
+        return database(node, validTime, transactionTime);
+    }
+
+    public Document document(Object contentHash) {
+        Map<Keyword,?> doc = node.document(contentHash);
+        return Document.document(doc);
+    }
+
+    public List<Document> history(CruxId id) {
+        List<Map<Keyword,?>> history = node.history(id.toEdn());
+        return history.stream()
+            .map(historyMap -> Document.document(historyMap))
```

Comment:
This seems wrong to me!

According to the published documentation, Crux returns a stream of _hashes_, so each Map is like
```
{:crux.db/id ; sha1 hash of document id
  "c7e66f757f198e08a07a8ea6dfc84bc3ab1c6613",
  :crux.db/content-hash ; sha1 hash of document contents
  "6ca48d3bf05a16cd8d30e6b466f76d5cc281b561",
  :crux.db/valid-time #inst "2018-05-18T09:20:27.966-00:00",
  :crux.tx/tx-time #inst "2019-04-15T07:53:55.817-00:00",
  :crux.tx/tx-id 1555314835817}
```

It seems that `node.history` returns a Map<Keyword, Object>, but it is **not a `Document`**!

Reply:
It might return an instance of a new class `DocumentSnapshot`, which has times, transaction id, and a document itself (loaded lazily upon request from its contentHash)

Reply:
cheers, fixed

Reply:
fixed :)